controllers/glanceapi_controller.go

```diff
@@ -205,6 +205,26 @@ func (r *GlanceAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *GlanceAPIReconciler) reconcileDelete(ctx context.Context, instance *glancev1beta1.GlanceAPI, helper *helper.Helper) (ctrl.Result, error) {
 	r.Log.Info("Reconciling Service delete")
 
+	// It's possible to get here before the endpoints have been set in the status, so check for this
+	if instance.Status.APIEndpoints != nil {
+		ksSvcSpec := keystonev1beta1.KeystoneServiceSpec{
+			ServiceType:        glance.ServiceType,
+			ServiceName:        glance.ServiceName,
+			ServiceDescription: "Glance Service",
+			Enabled:            true,
+			APIEndpoints:       instance.Status.APIEndpoints,
+			ServiceUser:        instance.Spec.ServiceUser,
+			Secret:             instance.Spec.Secret,
+			PasswordSelector:   instance.Spec.PasswordSelectors.Service,
+		}
```

Comment:
Rather than explicitly listing all the parameters here if you do need this you could simply lookup the KeystoneService by name right?